storm-core/src/jvm/backtype/storm/scheduler/resource/ResourceAwareScheduler.java

```diff
@@ -28,94 +32,270 @@
-            } else {
-                cluster.setStatus(td.getId(), "Fully Scheduled");
+            }
+        } else {
+            LOG.warn("Topology {} is already fully scheduled!", td.getName());
+            topologySubmitter.moveTopoFromPendingToRunning(td);
+            if (this.cluster.getStatusMap().get(td.getId()) == null || this.cluster.getStatusMap().get(td.getId()).equals("")) {
+                this.cluster.setStatus(td.getId(), "Fully Scheduled");
             }
         }
-        updateSupervisorsResources(cluster, topologies);
+    }
+
+    private boolean mkAssignment(TopologyDetails td, Map<WorkerSlot, Collection<ExecutorDetails>> schedulerAssignmentMap) {
+        if (schedulerAssignmentMap != null) {
+            double requestedMemOnHeap = td.getTotalRequestedMemOnHeap();
+            double requestedMemOffHeap = td.getTotalRequestedMemOffHeap();
+            double requestedCpu = td.getTotalRequestedCpu();
+            double assignedMemOnHeap = 0.0;
+            double assignedMemOffHeap = 0.0;
+            double assignedCpu = 0.0;
+
+            Set<String> nodesUsed = new HashSet<String>();
+            for (Map.Entry<WorkerSlot, Collection<ExecutorDetails>> workerToTasksEntry : schedulerAssignmentMap.entrySet()) {
+                WorkerSlot targetSlot = workerToTasksEntry.getKey();
+                Collection<ExecutorDetails> execsNeedScheduling = workerToTasksEntry.getValue();
+                RAS_Node targetNode = this.nodes.getNodeById(targetSlot.getNodeId());
+                targetNode.assign(targetSlot, td, execsNeedScheduling);
+                LOG.debug("ASSIGNMENT    TOPOLOGY: {}  TASKS: {} To Node: {} on Slot: {}",
+                        td.getName(), execsNeedScheduling, targetNode.getHostname(), targetSlot.getPort());
+                if (!nodesUsed.contains(targetNode.getId())) {
+                    nodesUsed.add(targetNode.getId());
+                }
+                assignedMemOnHeap += targetSlot.getAllocatedMemOnHeap();
+                assignedMemOffHeap += targetSlot.getAllocatedMemOffHeap();
+                assignedCpu += targetSlot.getAllocatedCpu();
+            }
+
+            Double[] resources = {requestedMemOnHeap, requestedMemOffHeap, requestedCpu,
+                    assignedMemOnHeap, assignedMemOffHeap, assignedCpu};
+            LOG.debug("setResources for {}: requested on-heap mem, off-heap mem, cpu: {} {} {} " +
+                            "assigned on-heap mem, off-heap mem, cpu: {} {} {}",
+                    td.getId(), requestedMemOnHeap, requestedMemOffHeap, requestedCpu,
+                    assignedMemOnHeap, assignedMemOffHeap, assignedCpu);
+            this.cluster.setResources(td.getId(), resources);
+            updateSupervisorsResources(this.cluster, this.topologies);
+            return true;
+        } else {
+            LOG.warn("schedulerAssignmentMap for topo {} is null. This shouldn't happen!", td.getName());
+            return false;
```

Comment:
I don't think so since if we get an invalid result from a strategy, we should ignore it instead of crashing nimbus.